query-node/mappings/src/workingGroups.ts

```diff
@@ -709,74 +721,110 @@ export async function workingGroups_StatusTextChanged({ store, event }: EventCon
   await store.save<StatusTextChangedEvent>(statusTextChangedEvent)
 }
 
-export async function workingGroups_LeadRemarked({ store, event }: EventContext & StoreContext): Promise<void> {
-  const [metadataByte] = new WorkingGroup_LeadRemarkedEvent_V1001(event).params
-  const group = await getWorkingGroupOrFail(store, event)
-
-  const metadata = deserializeMetadata(RemarkMetadataAction, metadataByte)
-  if (metadata?.moderatePost) {
+async function processWorkingGroupsRemark(
+  store: DatabaseManager,
+  event: SubstrateEvent,
+  group: WorkingGroup,
+  decodedMetadata: DecodedMetadataObject<IRemarkMetadataAction> | null,
+  isLead = false,
+  workerId: number
+): Promise<any> {
+  // Moderate Post
+  if (decodedMetadata?.moderatePost) {
```

Comment:
I'm not sure about refactor of WG remarks processing. I think it's more error-prone to mix lead and workers logic. Previously only leads could moderate posts, now you allowed all FWG workers to do it. You also use phantom `0` worker ID for leads which can lead to errors later on as well.

I suggest to keep current logic for forum moderation and verifying validators intact in `workingGroups_LeadRemarked`. Instead you can extract all the `if`s for tag messages into a function and then call it both in `LeadRemarked` and `WorkerRemarked`

Reply:
Yes. I will recover the original workingGroups.ts file and add the elseif branches for tagging system